nix/pkgs/aws-ofi-nccl/default.nix

```
@@ -0,0 +1,173 @@
+{
+  lib,
+  inputs,
+  self,
+  fetchFromGitHub,
+  symlinkJoin,
+  releaseTools,
+  stdenv,
+  config,
+  libfabric,
+  hwloc,
+  perl,
+  libtool,
+  autoconf,
+  automake,
+  autoreconfHook,
+  lttng-ust,
+  valgrind,
+  mpi,
+  cudaPackages ? { },
+  autoAddDriverRunpath,
+  neuronSupport ? (!config.cudaSupport),
+  cudaSupport ? (config.cudaSupport && !neuronSupport),
+  enableTests ? cudaSupport,
+  enableTracePrints ? true,
+  enableLTTNGTracing ? false,
+  enablePickyCompiler ? true,
+  enableWerror ? true,
+  enableNVTXTracing ? false,
+  enableValgrind ? false,
+  enableAwsTuning ? true,
+}:
+
+assert neuronSupport != cudaSupport;
+assert !enableNVTXTracing || (enableNVTXTracing && cudaSupport);
+let
+
+  effectiveStdenv = if cudaSupport then cudaPackages.backendStdenv else stdenv;
+
+  cudaBuildDepsJoined = symlinkJoin {
+    name = "cuda-build-deps-joined";
+    paths = lib.optionals (cudaSupport) (
```

```suggestion
    paths = lib.optionals cudaSupport (
```

Unnecessary parenthesis.